### PR TITLE
feat: adds labels to prompts

### PR DESCRIPTION
**Pull request by hassiebp**
Adds support for prompt labels

### CODE REVIEW

File: langfuse/model.py [hunk 1]
```diff
@@ -151,11 +158,13 @@ def get_langchain_prompt(self):
 class ChatPromptClient(BasePromptClient):
     def __init__(self, prompt: Prompt_Chat):
         super().__init__(prompt)
-        self.prompt = prompt.prompt
+        self.prompt = [
+            ChatMessageDict(role=p.role, content=p.content) for p in prompt.prompt
+        ]
 
-    def compile(self, **kwargs) -> List[ChatMessage]:
+    def compile(self, **kwargs) -> List[ChatMessageDict]:
```
Comment (maxdeichmann) [inline langfuse/model.py:165]: Is this a breaking change if someone imported `ChatMessage` originally? I think yes. The import path would have involved a path to the api module, correct?
If yes, this is hard to change, if no, i would suggest to keep the naming of the exported package.

File: langfuse/prompt_cache.py [hunk 2]
```diff
@@ -38,5 +38,19 @@ def set(self, key: str, value: PromptClient, ttl_seconds: Optional[int]):
...
+
+        if version is not None:
+            parts.append(f"version:{version}")
+
+        elif label is not None:
+            parts.append(f"label:{label}")
+
+        else:
+            # Default to production labeled prompt
+            parts.append("label:production")
```
Comment (maxdeichmann) [inline langfuse/prompt_cache.py:54]: this assumption makes sense.

File: tests/test_prompt.py [hunk 3]
```diff
@@ -167,6 +209,38 @@ def test_create_prompt_with_null_config():
...
+    assert default_prompt_client.version == 2
+    assert default_prompt_client.prompt == "test prompt 2"
+    assert default_prompt_client.labels == ["production"]
+
+    first_prompt_client = langfuse.get_prompt(prompt_name, 1)
+    assert first_prompt_client.version == 1
+    assert first_prompt_client.prompt == "test prompt 1"
+    assert first_prompt_client.labels == []
+
+    second_prompt_client = langfuse.get_prompt(prompt_name, version=2)
```
Comment (maxdeichmann) [inline tests/test_prompt.py:233]: image
Do you have the same signature when hovering?

File: langfuse/client.py [hunk 4]
```diff
@@ -538,6 +542,7 @@ def get_prompt(
         name: str,
         version: Optional[int] = None,
         *,
+        label: Optional[str] = None,
```
Comment (maxdeichmann) [inline langfuse/client.py:545]: would it make sense to add another overload like:


```
    @overload
    def get_prompt(
        self,
        name: str,
        *,
        version: Optional[int] = None,
        label: Optional[str] = None,
        type: Literal["text"] = "text",
        cache_ttl_seconds: Optional[int] = None,
    ) -> TextPromptClient: ...

```

File: tests/test_prompt.py (hunk 3 above)
Comment (marcklingen) [reply tests/test_prompt.py:233]: do we need `version` as `arg` for backwards compatibility while all others are `kwargs`?

File: langfuse/client.py [hunk 5]
```diff
@@ -648,7 +666,8 @@ def create_prompt(
         *,
         name: str,
         prompt: Union[str, List[ChatMessage]],
-        is_active: bool,
+        is_active: Optional[bool] = None,  # deprecated
+        labels: List[str] = [],
```
Comment (maxdeichmann) [inline langfuse/client.py:670]: as these are all named arguments, order does not matter - correct?

File: langfuse/prompt_cache.py (hunk 2 above)
Comment (marcklingen) [reply langfuse/prompt_cache.py:54]: agree, lgtm

**Review by marcklingen — APPROVED** [review]
looks good to me, awesome!

**Pull request merged**